test/integration/shoots/operations/containerruntime.go

```diff
@@ -35,10 +35,17 @@ var _ = ginkgo.Describe("Shoot container runtime testing", func() {
 		shoot := f.Shoot
 
 		if len(shoot.Spec.Provider.Workers) == 0 {
-			ginkgo.Skip("at least one worker pool is required in the test shoot.")
+			ginkgo.Skip("at least one worker pool is required in the test shoot")
 		}
 
-		containerdWorker := shoot.Spec.Provider.Workers[0].DeepCopy()
+		worker := shoot.Spec.Provider.Workers[0]
+		// containerD is supported only with Ubuntu OS for now.
+		// TODO: adapt/remove this when containerD is available on other OS.
+		if worker.Machine.Image.Name != "ubuntu" {
```

Comment:
Hm, @ialidzhikov I missed this and I don't like that this is hard-coded here. Does it make sense to make this configurable via command line flag which machine images are supported? Can you do that?

Reply:
Hmm, we also do it like that here:
https://github.com/gardener/gardener/blob/29da00ee28e332fb7348a55ebfbe3de1c1f49988/test/integration/shoots/operations/clusterautoscaler.go#L178-L180

How would this command line flag look like? Who would configure it / where will it be configured which tests should be enabled/skipped for which provider/os/...?
As @ialidzhikov mentioned:
> currently there is no mechanism from tm side to restrict test to given OS

I think, if we have such mechanism, this test would be a good fit for that.
@schrodit WDYT how to proceed on that?

Reply:
> Who would configure it / where will it be configured which tests should be enabled/skipped for which provider/os/...?

Somebody has to trigger the tests. Here those flags shall be configured.

Reply:
We have that problem on multiple tests.
I think we also have some aliclouds specifics.

I think we should have such mechanism that skip tests for specific providers or only run them for a small set of providers.

But I think that this should be configured in gardener so that test owners can directly configure the applicable shoot flavors when writing the tests.
With this we would have also simple versioning of that mapping file.
Maybe we can think of a specific configuration file that is evaluated by the testsuite and can run tests based on this config.

WDYT? @dguendisch

Reply:
I don't know, I'm not a fan of configuring such things in gardener/gardener. Instead, it should be configured in the extensions repos which tests they are supporting/not supporting. Otherwise, we are again having provider-specifics in g/g and this is something that we wanted to get rid of with extensibility...

Reply:
> Maybe we can think of a specific configuration file that is evaluated by the testsuite and can run tests based on this config.

Excellent idea! It avoids having to hard-code cloud-provider mappings.
When this file can be provided by the testmachinery, we could store such mappings along the mappings of flavors (in the landscape configuration).
I'd tend to prefer a more centralized storage of test-to-flavor/cloudprovider-applicability in our landscape configs as opposed to the test itself, simply because it gives us way better turnaround times in changing these configurations/mappings. So often we need to enable this test/flavor combination here or disable this test/flavor there because of some temporal thingy or because of a new flavor or test-dimension.

Reply:
A configuration that is supported by the gardener framework but reside in another repo to skip tests is valid.
Then the testmachinery would provide such config during runtime and we could maintain that file somewhere central.
This would give us the flexibility to skip tests in case of errors and temporary issues.

But regarding tests that only apply for specific cloudproviders.
I think we should have an inidication in the test itself that it is only supposed to work with a fixed set of cloudproviders.
Otherwise when the configuration is in a magic file outside g/g people do not know that these tests are only supposed to run on specific cloudproviders/flavors.

> Otherwise, we are again having provider-specifics in g/g and this is something that we wanted to get rid of with extensibility...

I get you're point. But then I these tests are provider specific and should be moved to the extension repos?

Reply:
> I think we should have an inidication in the test itself that it is only supposed to work with a fixed set of cloudproviders.
Otherwise when the configuration is in a magic file outside g/g people do not know that these tests are only supposed to run on specific cloudproviders/flavors.

I would try to flag such tests as "disabled by default" and add documentation that they have to explicitly enabled because they test stuff where provider-specific support/implementation is a prerequisite. Hence, the tests should only run if explicitly stated that the underlying extensions support the test case. Which of such tests are support and which not should be again maintained in the extensions repos.

> I get you're point. But then I these tests are provider specific and should be moved to the extension repos?

I wouldn't do that because it would result in a lot of duplication. What about above proposal?